cardano-testnet/src/Testnet/Runtime.hs

```diff
@@ -127,16 +127,18 @@ startNode tp node ipv4 port _testnetMagic nodeCmd = GHC.withFrozenCallStack $ do
 
   let nodeStdoutFile = logDir </> node </> "stdout.log"
       nodeStderrFile = logDir </> node </> "stderr.log"
+      nodePidFile = logDir </> node </> "node.pid"
       socketRelPath = socketDir </> node </> "sock"
       sprocket = Sprocket tempBaseAbsPath socketRelPath
 
   hNodeStdout <- retryOpenFile nodeStdoutFile IO.WriteMode
   hNodeStderr <- retryOpenFile nodeStderrFile IO.ReadWriteMode
+  hNodePid <- retryOpenFile nodePidFile IO.WriteMode
```

Comment:
`retryOpenFile` is too powerful here - we don't need it here. `IO.writeFile` should be enough and it would save you from having to deal with file handles.

We're using `retryOpenFile` because sometimes the operating system doesn't want to let go of file handles passed to `CreateProcess`, and when restarting the node, it results an error preventing the testnet from start. So it's a workaround for that rare, but annoying issue.